Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -319,15 +334,13 @@ void InsertPageBefore(Page page, Page before)
 			if (((INavigationPageController)this).StackDepth == 1)
 				return;
 
-			var root = (Page)PageController.InternalChildren.First();
-
-			var childrenToRemove = PageController.InternalChildren.ToArray().Where(c => c != root);
+			var childrenToRemove = PageController.InternalChildren.ToArray().Where(c => c != RootPage);
```

Comment:
Rather than using `Where()`, which will have to check the predicate for every page in the collection, could we use something like `var childrenToRemove = PageController.InternalChildren.Skip(1).ToArray();`?